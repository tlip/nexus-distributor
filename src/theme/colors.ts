interface IColorsType {
  [color: string]: string;
}

export const colors: IColorsType = {
  primary: '#54aa8b',
  secondary: '#f8fafd',
  tertiary: '#f0f3f6',
  black: '#3a3d42',
  white: '#ffffff',
  background: '#ffffff',
  blue: '#4fa7d9',
  yellow: '#e3be3c',
  red: '#D9684F',
  textPrimary: '#505050',
  textLight: 'rgba(80, 80, 80, 0.5)',
  textGray: '#7e8da5',
  textLink: '#0000ff',
};
