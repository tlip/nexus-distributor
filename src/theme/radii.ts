interface IRadiiScaleArray<T>
  extends IThemeScaleArray<
    T,
    {
      slight: number | string;
      small: number | string;
      medium: number | string;
      large: number | string;
      circle: string;
    }
  > {}

const radiiScale = [5, 10, 15, 20, '100%'];
const radiiScaleObject = {
  slight: radiiScale[0],
  small: radiiScale[1],
  medium: radiiScale[2],
  large: radiiScale[3],
  circle: radiiScale[4],
};

export const radii: IRadiiScaleArray<number | string> = Object.assign(
  radiiScale,
  radiiScaleObject
);
