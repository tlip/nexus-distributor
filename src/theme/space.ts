interface ISpaceScaleArray<T>
  extends IThemeScaleArray<T, Record<string, unknown>> {
  // ...to do? maybe?
}

const spaceScale = new Array(14).fill(0).map((_, i) => `${i * 0.25}em`);
const spaceScaleObject = {
  // if we need it?
};

export const space: ISpaceScaleArray<number | string> = Object.assign(
  spaceScale,
  spaceScaleObject
);
