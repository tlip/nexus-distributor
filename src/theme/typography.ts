interface ITypographyScaleArray<T>
  extends IThemeScaleArray<
    T,
    {
      h1: number | string;
      h2: number | string;
      h3: number | string;
      h4: number | string;
      h5: number | string;
      h6: number | string;
      subhead: number | string;
      body1: number | string;
      body2: number | string;
      button: number | string;
      caption1: number | string;
      caption2: number | string;
    }
  > {}

interface ITypography {
  fonts: { [variant: string]: string };
  fontWeights: { [variant: string]: number };
  fontSizes: ITypographyScaleArray<number | string>;
  letterSpacings: ITypographyScaleArray<number | string>;
  text: { [variant: string]: { [styleProp: string]: string | number } };
}

/**
 * System-UI font-size scale
 */
const fontSizesScale = [10, 12, 14, 16, 18, 20, 22, 24, 28, 34, 42];
const fontSizesScaleObject = {
  bigTitle: fontSizesScale[10],
  h1: fontSizesScale[9],
  h2: fontSizesScale[8],
  h3: fontSizesScale[6],
  h4: fontSizesScale[4],
  h5: fontSizesScale[3],
  h6: fontSizesScale[2],
  subhead: fontSizesScale[5],
  body1: fontSizesScale[3],
  body2: fontSizesScale[2],
  button: fontSizesScale[2],
  caption1: fontSizesScale[1],
  caption2: fontSizesScale[0],
  error: fontSizesScale[2],
};
const fontSizes: ITypographyScaleArray<number | string> = Object.assign(
  fontSizesScale,
  fontSizesScaleObject
);

/**
 * System-UI letter-spacing scale
 */
const letterSpacingsScale = [-1, -0.75, -0.5, -0.25, 0, 0.1, 0.15, 0.4, 1.5];
const letterSpacingsScaleObject = {
  bigTitle: letterSpacingsScale[0],
  h1: letterSpacingsScale[1],
  h2: letterSpacingsScale[2],
  h3: letterSpacingsScale[3],
  h4: letterSpacingsScale[4],
  h5: letterSpacingsScale[5],
  h6: letterSpacingsScale[6],
  subhead: letterSpacingsScale[5],
  body1: letterSpacingsScale[6],
  body2: letterSpacingsScale[6],
  button: letterSpacingsScale[3],
  caption1: letterSpacingsScale[7],
  caption2: letterSpacingsScale[8],
  error: letterSpacingsScale[6],
  mono: letterSpacingsScale[6],
};
const letterSpacings: ITypographyScaleArray<number> = Object.assign(
  letterSpacingsScale,
  letterSpacingsScaleObject
);

/**
 * Helper function to keep text variant definitions a bit more dry
 */
const createTextVariant = (
  variant: string,
  fontWeight: string,
  opts?: { [styleName: string]: string | number }
) => ({
  fontFamily: 'default',
  fontSize: variant,
  letterSpacing: variant,
  color: 'text.primary',
  px: 0,
  fontWeight,
  ...opts,
});

/**
 * Complete `typography` theme export
 */
export const typography: ITypography = {
  fonts: {
    default: "'Open Sans', sans-serif",
  },
  fontWeights: {
    regular: 400,
    semibold: 600,
    bold: 700,
  },
  fontSizes,
  letterSpacings,
  text: {
    bigTitle: createTextVariant('bigTitle', 'semibold'),
    h1: createTextVariant('h1', 'semibold'),
    h2: createTextVariant('h2', 'semibold'),
    h3: createTextVariant('h3', 'semibold'),
    h4: createTextVariant('h4', 'semibold'),
    h5: createTextVariant('h5', 'semibold'),
    h6: createTextVariant('h6', 'semibold'),
    subhead: createTextVariant('subhead', 'regular'),
    body1: createTextVariant('body1', 'regular'),
    body2: createTextVariant('body2', 'regular'),
    button: createTextVariant('button', 'semibold'),
    caption1: createTextVariant('caption1', 'regular'),
    caption2: createTextVariant('caption2', 'semibold', {
      textTransform: 'uppercase',
    }),
    error: createTextVariant('error', 'bold', {
      my: '0.8em',
      color: 'accents.red',
    }),
  },
};
