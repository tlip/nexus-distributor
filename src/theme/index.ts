import { colors } from './colors';
import { typography } from './typography';
import { radii } from './radii';
import { space } from './space';
import { buttons } from './variants/buttons';
import { cards } from './variants/cards';

export const theme: any = {
  space,
  breakpoints: ['0', '40em', '52em', '87.5em', '103.125em', '108.125em'],
  radii,
  colors,
  buttons,
  variants: {
    cards,
  },
  ...typography,
};
