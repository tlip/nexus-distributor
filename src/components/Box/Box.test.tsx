import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'theme-ui';
import { theme } from 'theme';
import { Box } from '.';

describe('Box component', () => {
  it('should render without crashing', async () => {
    const { container } = render(
      <ThemeProvider {...{ theme }}>
        <Box>I'm a box</Box>
      </ThemeProvider>
    );

    expect(await screen.findByText(/I'm a box/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
