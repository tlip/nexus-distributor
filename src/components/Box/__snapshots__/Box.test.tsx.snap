// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component should render without crashing 1`] = `
<div>
  <div
    class="css-1us58tu-Box e1gx7u5e0"
  >
    I'm a box
  </div>
</div>
`;
